the-collab-lab/tcl-6-smart-shopping-list
# sk-es-delete-items

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

This PR adds a trash can icon that allows users to remove any item from the shopping list. When the icon is clicked a confirmation window pops ups, this allows the user to verify that they really want to delete an item from the shopping list. We were able to accomplish this by wrapping our delete call to the database in a conditional statement. Which allows the item to not be immediately deleted without user consent. 

### Delete Items From The Database



## Related Issue
closes #13 


## Acceptance Criteria

* User is able to delete an item from the shopping list
* Prompt the user to confirm that they really want to delete the item to prevent accidental deletions
* Deletion should cause the associated record(s) in the database to be deleted
## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before
No trash can icon at the end of a shopping list item.
<img width="938" alt="Screen Shot 2020-05-13 at 3 47 31 PM" src="https://user-images.githubusercontent.com/33433415/81858070-2979b680-9531-11ea-8a82-9a9625c24178.png">




### After
CSS styling of the list item buttons was updated.
<img width="938" alt="Screen Shot 2020-05-13 at 3 46 35 PM" src="https://user-images.githubusercontent.com/33433415/81858059-24b50280-9531-11ea-8cee-71e599fe7175.png">




## Testing Steps / QA Criteria



[Netlify testing link for sk-es-delete-items](https://deploy-preview-40--romantic-lamarr-b840fb.netlify.app/) 
1. Create a new list or join an existing list using `deal focus mixup`.
2. Navigate to the Shopping view, if there are list items you will see a trash can icon at the end of the item's name. If you do not have any items on your shopping list use the Add Item view.
3. Clicking on the trash can icon will bring up a pop-up box. Confirming if you really want to delete that item. If you select `ok` it will be removed. If you select `cancel` the pop-up will close and the item will still remain on the shopping list.
